app/services/data_stage/process_school_changes.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module DataStage
+  class ProcessSchoolChanges < ::BaseService
+    include GiasTypes
+
+    UNHANDLED_ATTRIBUTES = %w[
+      school_type_code
+      school_type_name
+      section_41_approved
+      ukprn
+    ].freeze
+
+    def call
+      DataStage::SchoolChange.includes(:school).unhandled.status_changed.find_each do |change|
+        next if has_unhandled_changes?(change)
+
+        if change.attribute_changes.key? "school_status_code"
+          handle_status_code_change(change)
```

Comment:
How do we detect mergers/splits?

Reply:
That is done in the `close_school` if we don't have a simple "Successor" then we bail

Reply:
I don't really follow. In a split, wouldn't there be two successors?

Reply:
And it a merger, each of the schools have a single successor (the same one)

Reply:
I have some reservations about this (quite possibly because I've misunderstood it). It seems to me that this will:

- Open the successor school even when we can't handle the change and close the predecessor
- Close the predecessor and mark the change as handled even when we cannot move the assets etc

Reply:
The intention is that it will always open the new school, but only close the old one if we can handle the change.

Reply:
Mergers and splits as far as I've been able to tell have different school link reasons in GIAS, not simply "Successor" it would be something like "Successor - merged"

Reply:
I don't think the change gets marked as handled on a close if we can't perform the change?